web/src/main/java/com/google/sps/servlets/SaveStockServlet.java

```diff
@@ -45,22 +49,57 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
     KeyFactory keyFactory = datastore.newKeyFactory().setKind("Stock");
+    ArrayList<String> tikkers = new ArrayList<String>();
+    for (int i = 0; i < tik.size(); i++) {
+      String tikk = tik.get(i).text();
+      tikkers.add(tikk);
+    }
+
 
     for (int i = 0; i < tik.size(); i++) {
+      Key myTiks = datastore.newKeyFactory().setKind("Stock").newKey(tikkers.get(i));
 
-      String tikk = tik.get(i).text();
       String pricee = price.get(i).text().replaceAll("[\\\\$,]", "");
-      double priceDouble = Double.parseDouble(pricee);
+      Double priceDouble = Double.parseDouble(pricee);
 
-      FullEntity taskEntity =
-          Entity.newBuilder(keyFactory.newKey())
-              .set("Tik", tikk)
-              .set("TimeStamp", timeStamp)
+      Entity taskEntity =
+          Entity.newBuilder(datastore.get(myTiks))
+              .set("Tik", tikkers.get(i))
```

Comment:
"tikkers" spelt wrong, "tickers.".

Reply:
I know the spelling haha, I was just trying random things and was putting whatever came to my head, Sorry about that.